### PR TITLE
feat(clean): add clean_currency function

### PR DESCRIPTION
**Pull request by peterirani**
# Description

Add clean_currency function. 
This is WIP and is only used to go over the current design and test cases.  Some features still need to be implemented. 
This passes all the local tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have already squashed the commits and make the commit message conform to the project standard.
- [x] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: dataprep/clean/clean_currency.py [hunk 1]
```diff
@@ -0,0 +1,547 @@
...
+    split: bool = False,
+    inplace: bool = False,
+    report: bool = True,
+    progress: bool = False,
+) -> Union[pd.DataFrame, dd.DataFrame]:
+
+    """
+    Clean and standardize IP addresses.
+
+    Read more in the :ref:`User Guide <ip_userguide>`.
```
Comment (yxie66) [inline dataprep/clean/clean_currency.py:57]: The comment can be adjusted to be consistent with the function

**Comment by yxie66** [comment]
Hi @peterirani , thanks a lot for creating this. I have reviewed it and think that it's ready to get merged, except minor fix on comment. If you have test cases ready, please also feel free to create another PR, then I can test them and approve both PR together. Do you know why this PR doesn't pass Codacy check?

**Review by peterirani — COMMENTED** [review]
Hi @yxie66, I have made the necessary changes and added an example. Can you please go through it and let me know. The code doesn't pass on Codacy because of an `http.client` module warning. 

http_client_error

**Comment by yxie66** [comment]
> Hi @yxie66, I have made the necessary changes and added an example. Can you please go through it and let me know. The code doesn't pass on Codacy because of an `http.client` module warning.
> 
> http_client_error

Thank you Peter! Can you squash and rebase your branch to develop? Then I can approve the PR.

**Review by yxie66 — APPROVED** [review]

**Pull request merged**